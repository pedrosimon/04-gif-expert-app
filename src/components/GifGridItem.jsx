import React from "react";
import PropTypes from "prop-types";

const GifGridItem = ({ img }) => {
  const { title, url } = img;
  return (
    <div className="card animate__animated animate__bounce">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifGridItem.propTypes = {
  img: PropTypes.object,
};

export default GifGridItem;
