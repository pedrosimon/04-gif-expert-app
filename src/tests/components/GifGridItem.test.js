import React from 'react';
import {shallow} from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en el GifGridItem', () => {
    const titulo = 'Un Titulo';
    const url = 'https://algo.com';
    const claseNombre = 'animate__bounce';
    
    let wrapper = shallow(<GifGridItem img={{title: titulo, url: url}} />);

    beforeEach(() => {
        wrapper = shallow(<GifGridItem img={{title: titulo, url: url}} />);
    })
    
    test('debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe tener un pÃ¡rrafo con el Titulo', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(titulo);
    })
    
    test('debe de tener la imagen con alt y el url', () => {
        const img = wrapper.find('img');
        const {alt, src} = img.props();
       
        expect(alt).toBe(titulo);
        expect(src).toBe(url)
    })

    test('Debe tener la clase animate__bounce', () => {
        const div = wrapper.find('div');
        const {className} = div.props();
        // const clases = div.props('className');
        const clase = className.split(' ')[2];

        expect(clase).toBe(claseNombre);
        expect(div.hasClass(claseNombre)).toBe(true);
        expect(className.includes(claseNombre)).toBe(true);
    })
})
