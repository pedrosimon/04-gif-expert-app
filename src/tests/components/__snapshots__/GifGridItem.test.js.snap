// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Un Titulo"
    src="https://algo.com"
  />
  <p>
    Un Titulo
  </p>
</div>
`;
